### PR TITLE
STYLE make black local hook run twice as fast

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Since https://ichard26.github.io/blog/2022/05/compiling-black-with-mypyc-part-1/ , `black` runs twice as fast when the wheels are installed, compared with when it's built from source in the pre-commit hook

---

EDIT: CI timings show 53.84s vs 131.85s, so this is more than a 2x improvement

### CODE REVIEW

File: .github/workflows/code-checks.yml [hunk 1]
```diff
@@ -36,6 +36,8 @@ jobs:
 
     - name: Run pre-commit
       uses: pre-commit/action@v2.0.3
+      with:
+        extra_args: --verbose --all-files
```
Comment (MarcoGorelli) [inline .github/workflows/code-checks.yml:40]: `verbose` just shows the timings for each hook. `--all-files` is the default `extra_args`

File: environment.yml [hunk 2]
```diff
@@ -85,7 +85,7 @@ dependencies:
   - cxx-compiler
 
   # code checks
-  - black=22.3.0
+  - black=22.10.0
```
Comment (MarcoGorelli) [inline environment.yml:88]: at some point I'll address https://github.com/pandas-dev/pandas/issues/46561

**Comment by jbrockmendel** [comment]
Just checking: this doesn’t assume anything about my dev encironment?

**Comment by MarcoGorelli** [comment]
> Just checking: this doesn’t assume anything about my dev encironment?

that's right!

**Comment by akx** [comment]
This could maybe also be implemented by adding `black==22.10.0` to the `additional_dependencies` of the nonlocal `black` hook, so precommit would also install the mypyced module...

**Comment by MarcoGorelli** [comment]
> This could maybe also be implemented by adding `black==22.10.0` to the `additional_dependencies` of the nonlocal `black` hook, so precommit would also install the mypyced module...

that doesn't work unfortunately, I get
```
[INFO] Installing environment for https://github.com/python/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/marcogorelli/.cache/pre-commit/repom_n0drrc/py_env-python3.8/bin/python', '-mpip', 'install', '.', 'black==22.10.0')
return code: 1
expected return code: 0
stdout:
    Processing /home/marcogorelli/.cache/pre-commit/repom_n0drrc
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
    
    The conflict is caused by:
        The user requested black 0.1.dev1+g27d2014 (from /home/marcogorelli/.cache/pre-commit/repom_n0drrc)
        The user requested black==22.10.0
    
    To fix this you could try to:
    1. loosen the range of package versions you've specified
    2. remove package versions to allow pip attempt to solve the dependency conflict
    
    
stderr:
    ERROR: Cannot install black 0.1.dev1+g27d2014 (from /home/marcogorelli/.cache/pre-commit/repom_n0drrc) and black==22.10.0 because these package versions have conflicting dependencies.
    ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
    
Check the log at /home/marcogorelli/.cache/pre-commit/pre-commit.log
```

**Comment by akx** [comment]
@MarcoGorelli Ah, I guess black uses setuptools_scm or something so non-tagged versions are always dev+stuff (which makes sense). Thanks for giving it a shot!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**